### ISSUE TITLE
Support setting application ID for Wayland

### ISSUE THREAD

**Issue by Detegr** [emilk/egui#1600]
**Is your feature request related to a problem? Please describe.**
Applications using `egui` on Wayland have their [app_id](https://wayland.app/protocols/xdg-shell#xdg_toplevel:request:set_app_id) property set as `null`. As far as I can tell there is currently no way to set it.

**Describe the solution you'd like**
I'd like to have the window title to be used as `app_id` for Wayland. This looks to be as simple as calling the WindowBuilder's [with_app_id](https://docs.rs/winit/latest/winit/platform/unix/trait.WindowBuilderExtUnix.html#tymethod.with_app_id) function with the title string as the value.

**Additional context**
Some window managers, `sway` to name one, can make use of `app_id` value to detect windows of certain applications, and handle those windows in some different manner than others.

I'm willing to submit a pull request implementing this if this is something you agree would be an useful feature to have.

**Comment by dimtpap** [comment]
This would also allow some DEs/WMs to show an icon for the application in various places such as the title bar, if the application has provided desktop configuration files and icons

**Comment by Detegr** [comment]
> I assume the hard part is to configure this source to only run on Wayland (I'm not sure if this is exposed to the eframe crate as configuration yet).

`eframe` has a `wayland` feature so you should be able to use `#[cfg(feature = "wayland")]`.

> I am also not sure if it is wise to re-use the application title here (which is the same as the application name currently, but will this always be the case?), or if a more specific option e.g. in `eframe::NativeOptions` would better.

I'm not sure about this either.

**Comment by dimtpap** [comment]
> I am also not sure if it is wise to re-use the application title here (which is the same as the application name currently, but will this always be the case?), or if a more specific option e.g. in `eframe::NativeOptions` would better.

App IDs are supposed to be in reverse domain format like `org.example.Application` but the title would usually be just `Application` so a more specific option would be better. [Relevant documentation](https://wayland.app/protocols/xdg-shell#xdg_toplevel:request:set_app_id)

**Comment by thomaskrause** [comment]
I added a pull request (https://github.com/emilk/egui/pull/3007) for this, but decided against using the title as default. This would alter the behavior of existing programs, and I am uncertain if there are any limitations to application IDs that would conflict with more "weird" window titles. I used the `NativeOptions` instead, since there is already a macOS specific configuration value and so it makes sense to add a Wayland specific one there.
